learn/how-tos/user-driven-dynamic-roles.md

```diff
@@ -0,0 +1,103 @@
+---
+title: "User-Driven Dynamic Roles"
+id: ""
+---
+---
+
+Role-based Access Control (RBAC) is essential to restrict end-users from accessing specific sources and actions, including Pages, Widgets, and Services. It is easy to manage application roles for small and single-page applications. However, complex applications can have over a hundred permissions, which is not quite simple. 
+
+## Key RBAC Challenges
+
+1. Generally, the roles are managed and assigned during the application design time on WaveMaker.
+2. To add a new role to the application, the code needs to be ch  anged and re-deploy.
+3. For instance, when working with large Enterprise applications, app administrators typically have to manage multiple app users and roles.
+
+### How this document helps 
+
+1. Handling complex application permissions by defining application roles dynamically without many code changes.
```

Comment:
Instead of the word complex, we should use a different word. dynamic, or roles determined at runtime in an external system?

Reply:
@prasanth-reddy-wavemaker-com @nageshl 
1. Handling dynamic application permissions by defining the roles determined at runtime in an external system. 

**external system**: what is an external system? Should we be more specific?